Sources/DeepLearning/Optimizer.swift

```diff
@@ -84,11 +84,11 @@ public class RMSProp<Model: Layer, Scalar: TensorFlowFloatingPoint>: Optimizer
     public let decay: Scalar
 
     public init(
+        for _: __shared Model,
         learningRate: Scalar = 0.001,
         rho: Scalar = 0.9,
         epsilon: Scalar = 1e-8,
         decay: Scalar = 0,
-        modelType: Model.Type = Model.self,
         scalarType: Scalar.Type = Scalar.self
```

Comment:
Actually adding back the `scalarType` argument is not good enough to eliminate unexpected behavior. I'd suggest removing the default value so that `learningRate` won't make this argument be inferred.

Reply:
Removed default value. Thanks!